### PR TITLE
TEST: getCapturedRequestsAndClear should be atomic

### PR DESCRIPTION
**Pull request by dnhatn**
We might lose messages between getCapturedRequestsAndClear calls. This
commit makes sure that both getCapturedRequestsAndClear and
getCapturedRequestsByTargetNodeAndClear are atomic.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/transport/CapturingTransport.java [hunk 1]
```diff
@@ -94,26 +95,31 @@ public CapturedRequest(DiscoveryNode node, long requestId, String action, Transp
...
-        return capturedRequests;
+        List<CapturedRequest> requests = new ArrayList<>(capturedRequests.size());
+        capturedRequests.drainTo(requests);
+        return requests.toArray(new CapturedRequest[0]);
+    }
+
+    private Map<String, List<CapturedRequest>> groupRequestsByTargetNode(Collection<CapturedRequest> requests) {
+        Map<String, List<CapturedRequest>> result = new HashMap<>();
+        for (CapturedRequest request : requests) {
+            result.compute(request.node.getId(), (k, group) -> {
```
Comment (jasontedor) [inline test/framework/src/main/java/org/elasticsearch/test/transport/CapturingTransport.java]: I think this is simpler?

```diff
diff --git a/test/framework/src/main/java/org/elasticsearch/test/transport/CapturingTransport.java b/test/framework/src/main/java/org/elasticsearch/test/transport/CapturingTransport.java
index a7a0eb446e8..c8f87e75e15 100644
--- a/test/framework/src/main/java/org/elasticsearch/test/transport/CapturingTransport.java
+++ b/test/framework/src/main/java/org/elasticsearch/test/transport/CapturingTransport.java
@@ -103,13 +103,7 @@ public class CapturingTransport implements Transport {
     private Map<String, List<CapturedRequest>> groupRequestsByTargetNode(Collection<CapturedRequest> requests) {
         Map<String, List<CapturedRequest>> result = new HashMap<>();
         for (CapturedRequest request : requests) {
-            result.compute(request.node.getId(), (k, group) -> {
-                if (group == null) {
-                    group = new ArrayList<>();
-                }
-                group.add(request);
-                return group;
-            });
+            result.computeIfAbsent(request.node.getId(), k -> new ArrayList<>()).add(request);
         }
         return result;
     }
```

**Review by jasontedor — APPROVED** [review]
LGTM. I left one suggestion.

File: test/framework/src/main/java/org/elasticsearch/test/transport/CapturingTransport.java (hunk 1 above)
Comment (dnhatn) [reply test/framework/src/main/java/org/elasticsearch/test/transport/CapturingTransport.java]: +1

**Comment by dnhatn** [comment]
Thanks @jasontedor

**Pull request merged**